### ISSUE TITLE
FATAL error during execution of JavaScript file 'server/initialize.js' after upgrade to 3.1.6

### ISSUE THREAD

**Issue by mrkvon** [arangodb/arangodb#2248]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] self-compiled devel branch (3.1.6 - via [Arch Linux AUR package](https://aur.archlinux.org/packages/arangodb))

On this operating system:
- [x] Linux 
  - [x] other: Arch Linux

### this is an installation-related issue:
After upgrade from v3.0.7 to 3.1.6 via [Arch Linux AUR package](https://aur.archlinux.org/packages/arangodb) running
```bash
arangod
```
or
```bash
arangod --database.auto-upgrade
```
arangodb fails to start with the following output:
```
2017-01-04T13:27:48Z [1968] INFO ArangoDB 3.1.6 [linux] 32bit, using VPack 0.1.30, ICU 54.1, V8 5.0.71.39, OpenSSL 1.0.2j  26 Sep 2016
2017-01-04T13:27:48Z [1968] INFO using SSL options: SSL_OP_CIPHER_SERVER_PREFERENCE, SSL_OP_TLS_ROLLBACK_BUG
2017-01-04T13:27:48Z [1968] INFO file-descriptors (nofiles) hard limit is 4096, soft limit is 1024
2017-01-04T13:27:48Z [1968] INFO Authentication is turned on
2017-01-04T13:27:48Z [1968] INFO Authentication system only
2017-01-04T13:27:48Z [1968] INFO Authentication for unix sockets is turned on
2017-01-04T13:27:50Z [1968] INFO JavaScript using startup '/usr/share/arangodb3/js', application '/var/lib/arangodb3-apps'
2017-01-04T13:27:54Z [1968] WARNING reached heap-size limit, interrupting V8 execution (heap size limit 36700160, used 1974912)
2017-01-04T13:27:54Z [1968] FATAL error during execution of JavaScript file 'server/initialize.js'
2017-01-04T13:27:54Z [1968] WARNING reached heap-size limit, interrupting V8 execution (heap size limit 36700160, used 1974740)
2017-01-04T13:27:54Z [1968] FATAL error during execution of JavaScript file 'server/initialize.js'
```

This didn't help:
```bash
arangod --javascript.v8-options="--max_old_space_size=4096"
```

but this solved the problem:
```bash
arangod --javascript.v8-options="--max_old_space_size=1024"
```

I got to this solution by a lucky trial and error, and i don't know what is the cause. The solution is rather counter-intuitive for me, since it involves not increasing, but decreasing heap-size limit. It wasn't happening before the mentioned upgrade. (I also upgraded arch linux just before upgrading Arango, so that could be another potential source of fail.)

If you think it is Arango-related issue and you wish to spend time with it: Do you need some more information?

**Comment by jsteemann** [comment]
I think this is because you're running it on a 32 bit OS.
The default value for max_old_space_size is 3 GB, which is potentially too high.

Does the following patch fix the startup?
```diff
diff --git a/lib/Basics/operating-system.h b/lib/Basics/operating-system.h
index 396ad6c..0d029d4 100644
--- a/lib/Basics/operating-system.h
+++ b/lib/Basics/operating-system.h
@@ -53,6 +53,8 @@
 
 #if defined(__arm__) || defined(__aarch64__)
 #define TRI_V8_MAXHEAP 1 * 1024
+#elif TRI_PADDING_32
+#define TRI_V8_MAXHEAP 1 * 1024
 #else
 #define TRI_V8_MAXHEAP 3 * 1024
 #endif
```

**Comment by mrkvon** [comment]
@jsteemann Yes. The patch fixes it. :+1:

**Comment by jsteemann** [comment]
Just pushed the patch in the 3.1 branch. Will become part of 3.1.8.

**Comment by dothebart** [comment]
Hi, thanks for your kind words - good to hear ;-)
We just released 3.1.8 with this bugfix.
